src/chainNode.ts

```diff
@@ -44,4 +62,57 @@ export default class ChainNode {
     const members = (await this.api.query.membership.members()).toJSON()
     return listParser.parse(members)
   }
+
+  async getAttachmentApiAddresses(account: string): Promise<OrgData[]> {
+    await this.api.isReady
+    // Call the organisationData RPC method with the appropriate parameters
+    const attachmentKey = 'AttachmentEndpoint'
+    const oidcKey = 'OidcConfigurationEndpoint'
+    const decodedLiterals: OrgData[] = []
+    const urlRegex = /^(https?:\/\/[^\s/$.?#].[^\s]*)$/i
```

Comment:
instead of using a URL regex it might be better to see if you can parse the URL using the built in constructor https://developer.mozilla.org/en-US/docs/Web/API/URL/URL. I belive that throws if the URL is invalid